keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/ResourceId.kt

```diff
@@ -25,7 +25,12 @@ import com.netflix.spinnaker.keel.serialization.ResourceIdDeserializer
 data class ResourceId(val value: String) {
   override fun toString(): String = value
 
-  // Resource names (the last token in a ResourceId) follow the Moniker format (<app>-<stack>-<detail>)
-  val application: String
-    get() = value.split(":").last().split("-").first()
+  // Resource names (the last token in a ResourceId) follow the Moniker format (<app>-<stack>-<detail>),
+  // unless it's a resource which doesn't support this, then we return null (like an image)
+  val application: String?
+    get() = if (value.startsWith("bakery:")) {
+      null
+    } else {
+      value.split(":").last().split("-").first()
```

Comment:
Is it possible that the resource Id will not be in this format (<app>-<stack>-<detail>) or doesn't starts with "bakery:"? If so, can we get an exception here?

Reply:
I think the hard thing is there's no way to know if it doesn't fit the form except when you're implementing the resource. Right now, it's either it fits or it's the image. In the future, this could be different. I know this isn't the best way to do things, but I can't think of another way that keeps the convenience of this method (except removing the method and defining it only in resources that have a last piece of the id that's frigga named). Got any ideas?

Reply:
It looks to me like you removed the only place this was used `ApplicationVeto`. The two remaining uses are in `ResourceActuator` where it has access to the full resource & therefore could just use `resource.spec.application` instead.

I guess what I'm saying is, I think we can remove this property.

Reply:
Good call, removed and put a note on it to avoid parsing application from it (because I have a feeling that'll come up again...)